### PR TITLE
Migrate old Quick Start UI code to ImprovedMySiteFragment as Quick Start Block

### PR DESCRIPTION
**Pull request by ashiagr**
Issue #15127

This PR migrates old `Quick Start` UI code from `MySiteFragment` to `ImprovedMySiteFragment` as a `Quick Start Block`.

To test:

1. Go to `My Site` -> `App Settings` -> `Test feature configuration`.
2. Disable `QuickStartDynamicCardsFeatureConfig`, enable `MySiteImprovementsFeatureConfig`
3. Restart the app
4. Select a site with quick start in progress 
5. Notice that the old quick start section gets displayed according to completed/ uncompleted steps

qs_improved_my_site

## Regression Notes
1. Potential unintended areas of impact
    - `ImprovedMySiteFragment` -> `Quick Start - Dynamic Card` (`MySiteImprovementsFeatureConfig` - ON, `QuickStartDynamicCardsFeatureConfig` - ON )
    - `MySiteFragment` -> `Quick Start` (`MySiteImprovementsFeatureConfig` - OFF)

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually.

3. What automated tests I added (or what prevented me from doing so)
Added unit tests for `Quick Start Block` display.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt [hunk 1]
```diff
@@ -252,7 +254,7 @@ class MySiteViewModel
             }.associateBy { it.dynamicCardType }
 
             if (!quickStartDynamicCardsFeatureConfig.isEnabled() &&
-                    quickStartUtilsWrapper.isQuickStartInProgress(site.siteId.toInt())) {
+                    quickStartUtilsWrapper.isQuickStartInProgress(appPrefsWrapper.getSelectedSite())) {
```
Comment (zwarm) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt:257]: @ashiagr - DynamicCards source uses SiteRepository for siteId which monitors the site selection change.  I wonder why MySiteFragment uses AppPrefs?  🤔  Makes me question if siteId was wrong for those 1 and 1/2 months that ImprovedMySiteFragment was out OR maybe a matter of everything so intertwined.

**Review by zwarm — COMMENTED** [review]
@ashiagr  - great start and it was easy to review since you walked me through most of it this morning. Nice separation into the builder and +1 on the elimination of the store from the adapter. The "siteId" use is my only question... I called it out separately.

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt (hunk 1 above)
Comment (ashiagr) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt:257]: Yep, it is so much intertwined 🙂. As discussed, added it as an investigation task in the [parent issue](#15127).

**Review by zwarm — APPROVED** [review]
All good, thanks for taking care of Lint. Yes, siteId is a much larger discussion. Thanks!!

**Pull request merged**